contracts/src/utils/WitnessTx.sol

```diff
@@ -106,13 +107,28 @@ library WitnessTx {
     /// @param txInfo Bitcoin transaction data.
     /// @param proof Bitcoin proof data.
     /// @return wTxHash Proven 32-byte transaction hash.
-    function validateWitnessProofAndDifficulty(
+    function validateWitnessProof(
         ILightRelay relay,
         uint256 txProofDifficultyFactor,
         WitnessInfo memory txInfo,
         WitnessProof memory proof
     ) internal view returns (bytes32 wTxHash) {
         wTxHash = validateWitnessProof(txInfo, proof);
+
+        // Checks that the header chain is valid with respect to the difficulty stored in the light relay
         BitcoinTx.evaluateProofDifficulty(relay, txProofDifficultyFactor, proof.coinbaseProof.bitcoinHeaders);
     }
+
+    function validateWitnessProof(
+        IFullRelayWithVerify relay,
+        uint256 txProofDifficultyFactor,
```

Comment:
Technically for the full relay "difficulty factor" doesn't make so much sense, but won't block the PR on this.